pkg/clients/aws/route53.go

```diff
@@ -123,13 +126,23 @@ func (c *awsClient) AnswerDNSChallenge(reqLogger logr.Logger, acmeChallengeToken
 		return "", err
 	}
 
-	baseDomain := cr.Spec.ACMEDNSDomain
-	if !strings.HasSuffix(baseDomain, ".") {
-		baseDomain = baseDomain + "."
+	dnsZones := hivev1.DNSZoneList{}
+	err = c.kubeClient.List(context.TODO(), &dnsZones, &client.ListOptions{Namespace: c.namespace})
+	if err != nil {
+		reqLogger.Error(err, err.Error())
+		return "", err
+	}
+
+	if len(dnsZones.Items) == 0 {
+		return "", fmt.Errorf("no dns zones found")
 	}
+	dnsZone := dnsZones.Items[0]
```

Comment:
This assumes you are only getting one zone back. Can there be multiple objects? Should we error if there is more than one?